### PR TITLE
 cleanup static variables in PFResolutionMap.cc

### PR DESCRIPTION
**Pull request by slava77**
the title covers it.

I'm daring to submit this web-edited without local compile check.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc [hunk 1]
```diff
@@ -306,30 +225,35 @@ double PFResolutionMap::minimum(double a,double b){
...
-  cPhi.resize(18,0);
-  cPhi[0]=2.97025;
-  for(unsigned i=1;i<=17;i++) cPhi[i]=cPhi[0]-2*i*pi/18;
+  constexpr double c0 = 2.97025;
+  constexpr std::array<double,18> cPhi {{c0, 
+					c0-twopiO18, c0-2*twopiO18, c0-3*twopiO18, c0-4*twopiO18,
+					c0-5*twopiO18, c0-6*twopiO18, c0-7*twopiO18, c0-8*twopiO18,
+					c0-9*twopiO18, c0-10*twopiO18, c0-11*twopiO18, c0-12*twopiO18,
+					c0-13*twopiO18, c0-14*twopiO18, c0-15*twopiO18, c0-16*twopiO18,
+					c0-17*twopiO18}};
```
Comment (VinInn) [inline RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc]: this compile as well
```
constexpr
std::array<float,18> cPhiGener() {
  std::array<float,18> cPhi{0};
  cPhi[0]=2.97025;
  for(unsigned i=1;i<=17;i++) cPhi[i]=cPhi[0]-2*i*M_PI/18;
  return cPhi;
}

constexpr
std::array<float,18> cPhi = cPhiGener();
```

Comment (slava77) [reply RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc]: nice. I will change.
Thanks.

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc (hunk 1 above)
Comment (slava77) [reply RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc]: so, this didn't work.
I guess it's c++17 and implies gcc 7.0.

Comment (VinInn) [reply RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc]: indeed. I tested with my version of gcc (8.1)

**Comment by slava77** [comment]
@cmsbuild please test

after this was rebased to drop the commit with constexpr functions

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/23590	2a51552
- technical; this gets rid of 7 static analyzer warnings
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**